demos/common/python/models/yolo.py

```diff
@@ -418,3 +418,105 @@ def xywh2xyxy(x):
+        self.image_blob_name, self.image_info_blob_name = self._get_inputs()
+
+        self._output_layer_names = sorted(self.net.outputs)
+        assert len(self.net.outputs) == 3, "Expected 3 output blobs"
+        self.bboxes_blob_name, self.scores_blob_name, self.indices_blob_name = self._parse_outputs()
+
+        self.n, self.c, self.h, self.w = self.net.input_info[self.image_blob_name].input_data.shape
+        assert self.c == 3, "Expected 3-channel input"
+
+        if isinstance(labels, (list, tuple)):
+            self.labels = labels
+        else:
+            self.labels = load_labels(labels) if labels else None
+
+        self.confidence_threshold = threshold
+
+    def _get_inputs(self):
+        image_blob_name = None
+        image_info_blob_name = None
+        for blob_name, blob in self.net.input_info.items():
+            if len(blob.input_data.shape) == 4:
+                image_blob_name = blob_name
+            elif len(blob.input_data.shape) == 2:
+                image_info_blob_name = blob_name
+            else:
+                raise RuntimeError('Unsupported {}D input layer "{}". Only 2D and 4D input layers are supported'
+                                   .format(len(blob.shape), blob_name))
+        if image_blob_name is None:
+            raise RuntimeError('Failed to identify the input for the image.')
+        return image_blob_name, image_info_blob_name
+
+    def _parse_outputs(self):
+        bboxes_blob_name = None
+        scores_blob_name = None
+        indices_blob_name = None
+        for name, layer in self.net.outputs.items():
+            if layer.shape[-1] == 3:
+                indices_blob_name = name
+            elif layer.shape[2] == 4:
+                bboxes_blob_name = name
+            elif layer.shape[1] == 80:
+                scores_blob_name = name
+            else:
+                raise RuntimeError("Expected shapes [:,:,4], [:,80,:] and [:,3] for outputs, but got {}, {} and {}"
+                                   .format(*[output.shape for output in self.net.outputs]))
+        assert self.net.outputs[bboxes_blob_name].shape[1] == self.net.outputs[scores_blob_name].shape[2], \
+            "Expected the same dimension for boxes and scores"
+        return bboxes_blob_name, scores_blob_name, indices_blob_name
+
+    def preprocess(self, inputs):
+        image = inputs
+        resized_image = resize_image_letterbox(image, (self.w, self.h))
+
+        meta = {'original_shape': image.shape,
+                'resized_shape': resized_image.shape}
+
+        preprocessed_image = resized_image.transpose((2, 0, 1))  # Change data layout from HWC to CHW
+        preprocessed_image = preprocessed_image.reshape((self.n, self.c, self.h, self.w))
+
+        dict_inputs = {self.image_blob_name: preprocessed_image}
+        if self.image_info_blob_name:
+            dict_inputs[self.image_info_blob_name] = [image.shape[0], image.shape[1]]
```

Comment:
```suggestion
        dict_inputs[self.image_info_blob_name] = [image.shape[0], image.shape[1]]
```
Looks like the `image_info_blob` always exists